cpu/atxmega/include/periph_cpu.h

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_atxmega
+ * @{
+ *
+ * @file
+ * @brief           CPU specific definitions for internal peripheral handling
+ *
+ * @author          Gerson Fernando Budke <nandojve@gmail.com>
+ */
+
+#ifndef PERIPH_CPU_H
+#define PERIPH_CPU_H
+
+#include <avr/io.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name   Length of the CPU_ID in octets
+ * @{
+ */
+#define CPUID_LEN               (11U)
+/** @} */
+
+/**
+ * @name   Interrupt level used to control nested interrupts
+ * @{
+ */
+typedef enum {
+    CPU_INT_LVL_OFF,    /**< Interrupt Disabled     */
+    CPU_INT_LVL_LOW,    /**< Interrupt Low Level    */
+    CPU_INT_LVL_MID,    /**< Interrupt Medium Level */
+    CPU_INT_LVL_HIGH,   /**< Interrupt High Level   */
+} cpu_int_lvl_t;
+/** @} */
+
+/**
+ * @brief   Available ports on the ATxmega family
+ */
+enum {
+    PORT_A,     /**< port A - 600 - 0 */
+    PORT_B,     /**< port B - 620 - 1 */
+    PORT_C,     /**< port C - 640 - 2 */
+    PORT_D,     /**< port D - 660 - 3 */
+    PORT_E,     /**< port E - 680 - 4 */
+    PORT_F,     /**< port F - 6A0 - 5 */
+    PORT_G,     /**< port G - 6C0 - 6 */
+    PORT_H,     /**< port H - 6E0 - 7 */
+    PORT_J,     /**< port J - 700 - 8 */
+    PORT_K,     /**< port K - 720 - 9 */
+    PORT_L,     /**< port L - 740 - A */
+    PORT_M,     /**< port M - 760 - B */
+    PORT_N,     /**< port N - 780 - C */
+    PORT_P,     /**< port P - 7A0 - D */
+    PORT_Q,     /**< port Q - 7C0 - E */
+    PORT_R,     /**< port R - 7E0 - F */
+    /* ... */
+    PORT_MAX,
+};
+
+/**
+ * @name    Power management configuration
+ * @{
+ */
+#define PM_NUM_MODES            (4)
+/** @} */
+
+/**
+ * @brief   Define the number of GPIO interrupts vectors for ATxmega CPU.
+ * @{
+ */
+#define GPIO_EXT_INT_NUMOF      (2 * PORT_MAX)
+/** @} */
+
+/**
+ * @brief   Override GPIO type
+ * @{
+ */
+#define HAVE_GPIO_T
+typedef uint16_t gpio_t;
+/** @} */
+
+/**
+ * @brief   Definition of a fitting UNDEF value
+ */
+#define GPIO_UNDEF              (0xffff)
+
+/**
+ * @brief   Define a CPU specific GPIO pin generator macro
+ *
+ * The ATxmega internally uses pin mask to manipulate all gpio functions.
+ * This allows simultaneous pin actions at any method call.  ATxmega specific
+ * applications can use ATXMEGA_GPIO_PIN macro to define pins and generic
+ * RIOT-OS application should continue to use GPIO_PIN API for compatibility.
+ *
+ * @{
+ */
+#define ATXMEGA_GPIO_PIN(x, y)  (((x & 0x0f) << 8) | (y & 0xff))
+#define GPIO_PIN(x, y)          ATXMEGA_GPIO_PIN(x, (1U << (y & 0x07)))
```

Comment:
```suggestion
#define GPIO_PIN(x, y)          ATXMEGA_GPIO_PIN(x, (1U << (y)))
```

It shouldn't be needed to sanitize the second parameter, as this is already done in `ATXMEGA_GPIO_PIN()`. But it could be useful to add braces around `y` in case someone does `GPIO_PIN(PORT_A, 1 + 2)`.